main.cpp

```diff
@@ -275,296 +300,281 @@ struct Bakery
+    void padPreset(float eCurve);
+    void staccatoPreset(float eCurve);
+    bool loopOver(float eAttackTime, float eDecayTime, float eReleaseTime);
 };
 
-/*
-Thing 8) lfo
-5 properties:
-    1) fade-in time (float)
-    2) frequency (double)
-    3) shape (int)
-    4) retrigger behaviour (int)
-    5) mono / poly switch (bool)
-3 things it can do:
-    1) reset
-    2) modulate
-    3) change shape
- */
+void Envelope::padPreset(float eCurve)
+{
+    std::cout << "Setting envelope to slow setting, with curve of: " << eCurve << std::endl;
+}
+void Envelope::staccatoPreset(float eCurve)
+{
+    std::cout << "Setting envelope to fast setting, with curve of: " << eCurve << std::endl;
+}
+bool Envelope::loopOver(float eAttackTime, float eDecayTime, float eReleaseTime)
+{
+    bool eSwitch = false;
+
+    if ((eAttackTime + eDecayTime + eReleaseTime) < 700.f)
+    {
+        eSwitch = true;
+        return true;
+    }
+
+    return false;
+}
 
 struct LFO
 {
-    //1) fade-in time (float)
     float fadeIn = 45.3f;
-    //2) frequency (double)
     double freq = 0.074;
-    //3) shape (int)
     int shape = 2;
-    //4) retrigger behaviour (int)
     int retrig = 3;
-    //5) mono / poly switch (bool)
     bool polyMode = false;
 
-    //1) reset
     void resetState();
-    //2) modulate
-    void modulateFreq(double freq, float modFreq, float intensity);
-    //3) change shape
-    void changeSpeed(int shape);
+    void modulateFreq(double lFreq, float modFreq, float intensity);
+    void changeSpeed(int lShape);
 };
 
-/*
-Thing 9) waveform
-5 properties:
-    1) amplitude (float)
-    2) length in samples (int)
-    3) color (std::string)
-    4) height (int)
-    5) width (int)
-3 things it can do:
-    1) zoom in
-    2) zoom out
-    3) draw audio signal
- */
+void LFO::resetState()
+{ 
+}
+
+void LFO::modulateFreq(double lFreq, float modFreq, float intensity)
+{
+    std::cout << "LFO frequency set to: " << lFreq << std::endl << "Modulation speed set to: " << modFreq << std::endl << "Modulation intensity set to: " << intensity << std::endl;
+}
+
+void LFO::changeSpeed(int lShape)
+{
+    if (lShape == 0)
+        std::cout << "Changing sinusoid LFO frequency." << std::endl;
+    else
+        std::cout << "Changing triangle LFO frequency." << std::endl;
+}
 
 struct Waveform
 {
-    //1) amplitude (float)
     float amplitude = 0.89f;
-    //2) length in samples (int)
     int lengthInSamples = 644112;
-    //3) color (std::string)
     std::string waveColor = "Yellow";
-    //4) height (int)
     int height = 220;
-    //5) width (int)
     int width = 720;
 
-    //1) zoom in
     void zoomIn(float zoomInAmount);
-    //2) zoom out
     void zoomOut(float zoomOutAmount);
-    //3) draw audio signal
-    void drawWaveform(float amplitude, int lengthInSamples, int height, int width);
+    void drawWaveform(float wAmplitude, int wLengthInSamples, int wHeight, int wWidth);
```

Comment:
Review our Slack discussion of using member variables in member functions directly. 
Change your argument list so that your function arguments are not duplicates of your member variables.
This can mean changing the function name (meaning the function purpose has changed) 
or it could mean changing the function arguments.
Remember, name things what they are or what they do.
- [ ] revise